ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/Time.scala

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.tests
+
+import java.time.Duration
+
+import com.daml.ledger.api.testtool.infrastructure.{LedgerSession, LedgerTest, LedgerTestSuite}
+
+import scala.concurrent.duration.DurationInt
+
+final class Time(session: LedgerSession) extends LedgerTestSuite(session) {
+
+  val pass =
+    LedgerTest("Advancing time should return the new time") { implicit context =>
+      for {
+        t1 <- time()
+        _ <- passTime(1.second)
```

Comment:
How does this test work for wall-clock time? Or is it missing a `skipIf`?

Reply:
It would throw. The problem is it's quite difficult to reliably detect if the time service is there. In the case of the DAML-on-X reference implementation, for example, the `TimeService` exists also on wall-clock time but it's not capable of doing time travel (which makes sense). So using `skipIf` would not work very reliably unless we signal the compliance with the `TimeService` via configuration/CLI, which could be effectively equivalent to simply specifying to include or exclude the `TimeIT` test. Would that be a feasible approach to the problem?

Reply:
You can see an example of at the commit where I introduced running the conformance tests on the in-memory sandbox (https://github.com/digital-asset/daml/pull/2282/commits/9cec58d6d8ab8f83593307f394d3a5d813f8e4a7).

Reply:
You can see an example of at the commit where I introduced running the conformance tests on the in-memory (https://github.com/digital-asset/daml/pull/2282/commits/9cec58d6d8ab8f83593307f394d3a5d813f8e4a7) and persistent sandbox (https://github.com/digital-asset/daml/pull/2282/commits/549e26411bd5e7dee2af74a7052c327666de2832).

Reply:
Okay, then let's keep this test as optional and running that test should be a conscious opt-in.